rfc/rfc017-distributed-network-grpc-v2.md

```diff
@@ -45,4 +54,270 @@ As specified by RFC015, the node MUST authenticate the peer's node DID as follow
+
+A node MUST make sure to only add transactions of which all previous transactions are present.
+
+#### 6.2.1 Broadcasting the state
+
+The `State` message is sent as response to various conditions of the gossip protocol (see §5).
+The `State` message MUST contain a new `conversationID`.
+The `State` message contains a `XOR` value and an `LC`.
+The `LC` value MUST equal the highest Lamport Clock value of all transaction references included in the `XOR` calculation.
+If no transactions are present, an all-zero `XOR` and `LC` of 0 is sent.
+
+This might look exactly like the `Gossip` message, but the type of response is different so it is a different type of message.
+
+#### 6.2.2 IBLT response
+
+When a peer receives a `State` message and the `XOR` differs, it SHOULD respond with a `TransactionSet` message.
+The sent `LC` value falls within the bounds of a page.
+The response IBLT MUST be calculated over the transactions leading up to and including that page.
+If the peer's highest Lamport Clock value is lower, it MUST use the IBLT covering the entire DAG.
+Next to the IBLT data, the `TransactionSet` message MUST contain the original `LC` value as `LC_req`, a new `LC` value indicating the highest LC value from the peer.
+This is a response type message so it MUST contain the `conversationID`.
+
+#### 6.2.3 Transaction List Query
+
+For every `TransactionSet` message a node receives, it MUST check if the `LC_req` value matches the `LC` value from the original request.
+
+The IBLT in the `TransactionSet` message contains every transaction of the peer in the LC range of `0-min(LC, LC_req)`. The node MUST lookup/compute the IBLT for this range and subtract it from the IBLT of the peer. Decoding the resulting IBLT will list the transaction refs the peer has and the local node misses. 
+These transactions MUST be queried by using the `TransactionListQuery` message (see §5.2.2). 
+
+If the decoding fails, the local node sends a new `State` message. 
+The IBLT subtraction used the range `0-min(LC, LC_req)`.
+The new request MUST be for one page lower.
+The `LC` value MUST be the `end` of that previous page. 
+The `XOR` value MUST be calculated over the transaction references in the range starting with 0 and ending with the new `LC` value.
+
+#### 6.2.4 Transaction Range Query
+
+If the IBLT from a `TransactionSet` message can be decoded, the local node MAY also send a `TransactionRangeQuery` message.
+This is only needed if the page containing the `LC` value of the `TransactionSet` message comes after the page containing `LC_req`. 
+This means the peer has additional transactions outside the IBLT range.
+If the `LC_req` value is in the latest page of the local node, it SHOULD query all pages leading up to the `LC` value.
+If the `LC_req` value is NOT in the latest page of the local node, then it MUST only query the next page.
+This last requirement prevents a node from querying the entire DAG while only some historic transactions are missing.
```

Comment:
And solves the case where there is a collision on a page.

Reply:
added that too